features/varnish/cache.feature

```diff
@@ -0,0 +1,60 @@
+@varnish
+Feature: As an site administrator I want my pages to be cached using Varnish
+
+    @admin
+    Scenario Outline: Content Items are cached for user when visited
+        Given I create "Folder" Content items in root in "eng-GB"
+            | name       | short_name |
+            | TestFolder | <itemName> |
+        And I am viewing the pages on siteaccess "site" as "<user>"
+        When I visit <itemName> on siteaccess "site"
+        And response headers contain
+            | Header  | Value |
+            | X-Cache | MISS  |
+        And I reload the page
+        Then I see correct preview data for "Folder" Content Type
+            | field | value      |
+            | title | <itemName> |
+        And response headers contain
+            | Header  | Value |
+            | X-Cache | HIT   |
+
+        Examples:
+            | user      | password | itemName                     |
+            | admin     | publish  | TestFolderShortNameAdmin     |
+            | anonymous |          | TestFolderShortNameAnonymous |
+
+    @admin
+    Scenario Outline: Cache is refreshed when item is edited
+        Given I create "Folder" Content items in root in "eng-GB"
+            | name       | short_name |
+            | TestFolder | <itemName> |
+        And I am viewing the pages on siteaccess "site" as "<user>" "<password>"
+        And I visit "<itemName>" on siteaccess "site"
+        And I reload the page
+        And I see correct preview data for "Folder" Content Type
+            | field | value      |
+            | title | <itemName> |
+        And response headers contain
+            | Header  | Value |
+            | X-Cache | HIT   |
+        When I edit "<itemName>" Content item in "eng-GB"
+            | short_name          |
+            | <itemNameAfterEdit> |
+        And I reload the page
+        # Give Varnish time to fetch the backend response
```

Comment:
This is tricky: if I understand correctly, on Content edit the cache is soft-purged - meaning that Varnish will keep serving the old cache until it it fetched from the backend again.

The first reload triggers the backend fetch, but if we follow immediately with another reload we will still be served the stale content - we need to wait until the fetch is completed.

On my local env 1 second was enough for the test to pass, I've chosen 5 for CI safety.